socket.go

```diff
@@ -232,8 +232,9 @@ func (socket *mongoSocket) InitialAcquire(serverInfo *mongoServerInfo, timeout t
 		stats.socketsExpired(+1)
 		debugf("socket: %p cannot use socket - max connection reuse time expired", socket)
 		socket.Unlock()
-		socket.Close()
-		return errors.New("Socket reuse time expired")
+		err := errors.New("Socket reuse time expired")
+		socket.kill(err, true)
```

Comment:
Please comment on why kill is used instead of close, including a link to `https://github.com/ChimeraCoder/mgo/blob/master/socket.go#L283` or similar from your original comment on the PR.

Also, is this actually a bug in mgo that we're working around, or were we using it incorrectly?